lib/IDE/SyntaxModel.cpp

```diff
@@ -528,21 +529,23 @@ static bool shouldTreatAsSingleToken(const SyntaxStructureNode &Node,
 
 std::pair<bool, ArgumentList *>
 ModelASTWalker::walkToArgumentListPre(ArgumentList *ArgList) {
+  Expr *ParentExpr = Parent.getAsExpr();
+  if (!ParentExpr)
+    return {true, ArgList};
```

Comment:
I don't think this can currently ever happen (as `ModelASTWalker` isn't ever called to walk directly over an ArgumentList), but if it did, I believe it would cause `ModelASTWalker::walkToArgumentListPost` to try and pop on an empty stack.

Should `ModelASTWalker::walkToArgumentListPost` perhaps avoid popping in the null parent case, and perhaps assert for a non-null parent that it matches the element being popped?

Reply:
Ah good point, nice catch - thanks. Good ol' matching pre/post